label_studio/io_storages/proxy_api.py

```diff
@@ -0,0 +1,178 @@
+import base64
+import logging
+from typing import Union
+from urllib.parse import unquote
+
+from core.feature_flags import flag_set
+from django.conf import settings
+from django.http import HttpRequest, HttpResponseRedirect
+from projects.models import Project
+from ranged_fileresponse import RangedFileResponse
+from rest_framework import status
+from rest_framework.permissions import IsAuthenticated
+from rest_framework.response import Response
+from rest_framework.views import APIView
+from tasks.models import Task
+
+from label_studio.io_storages.functions import get_storage_by_url
+
+logger = logging.getLogger(__name__)
+
+
+class ResolveStorageUriAPIMixin:
+    def resolve(self, request: HttpRequest, fileuri: str, instance: Union[Task, Project]) -> Response:
+        model_name = type(instance).__name__
+
+        if not instance.has_permission(request.user):
+            return Response(status=status.HTTP_403_FORBIDDEN)
+
+        # Attempt to base64 decode the fileuri
+        try:
+            fileuri = base64.urlsafe_b64decode(fileuri.encode()).decode()
+        # For backwards compatibility, try unquote if this fails
+        except Exception as exc:
+            logger.debug(
+                f'Failed to decode base64 {fileuri} for {model_name} {instance.id}: {exc} falling back to unquote'
+            )
+            fileuri = unquote(fileuri)
+
+        # Try to find storage by URL
+        project = None
+        if flag_set('fflag_optic_all_optic_1938_storage_proxy', user='auto'):
+            project = instance if isinstance(instance, Project) else instance.project
+            storage_objects = project.get_all_import_storage_objects
+            storage = get_storage_by_url(fileuri, storage_objects)
+            if not storage:
+                logger.error(f'Could not find storage for URI {fileuri}')
+                return Response(status=status.HTTP_404_NOT_FOUND)
+            # Not all storages support presigned URLs
+            if not hasattr(storage, 'presign'):
+                logger.error(f'Storage {storage} does not support presign URLs')
+                return Response(status=status.HTTP_404_NOT_FOUND)
+            presign = storage.presign
+        else:
+            presign = True
```

Comment:
why do we default to true here?

Reply:
I guess so that if the FF is not set, we default to the old behavior, correct?

Reply:
Yes, you are right, it's default for the old behavior.